packages/chopsticks/src/blockchain/index.ts

```diff
@@ -125,9 +140,18 @@ export class Blockchain {
       } else {
         const loadingBlock = (async () => {
           try {
-            const header = await this.api.getHeader(hash)
-            const block = new Block(this, Number(header.number), hash)
-            this.#registerBlock(block)
+            if (this.db) {
+              const blockData = await this.db.getRepository('Block').findOne({ where: { hash } })
+              if (blockData) {
+                const { number, hash, header, extrinsics } = blockData
+                const block = new Block(this, number, hash, undefined, { header, extrinsics })
```

Comment:
I guess we don't need to apply extrinsics and get storageDiff here

Reply:
is it necessary to add parent block here?

Reply:
we need parent block.
I think we need to save storageDiff as for example it is possible to use `dev_setStorage` to override them and without saving those, the chain will be different after restart, which is unexpected.